presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -207,20 +210,22 @@ else if (queryStateMachine.getQueryState() == QueryState.STARTING) {
         PartitioningHandle partitioningHandle = plan.getFragment().getPartitioning();
         if (partitioningHandle.equals(SOURCE_DISTRIBUTION)) {
             // nodes are selected dynamically based on the constraints of the splits and the system load
-            SplitSource splitSource = plan.getDataSource().get();
-            NodeSelector nodeSelector = nodeScheduler.createNodeSelector(splitSource.getDataSourceName());
+            Entry<PlanNodeId, SplitSource> entry = Iterables.getOnlyElement(plan.getSplitSources().entrySet());
```

Comment:
I'm very confused by this commit. Tell me if my understanding is correct:

* If a stage has 2 tablescan, it will be considered fixed partitioned (not source partitioned) because a hash function must exist that enables them to be colocated
* FixedCount never has source in it
* Source always has only 1